shell/platform/darwin/ios/framework/Source/FlutterViewControllerTest.mm

```diff
@@ -1479,6 +1452,33 @@ - (void)testMouseSupport API_AVAILABLE(ios(13.4)) {
       dispatchPointerDataPacket:std::make_unique<flutter::PointerDataPacket>(0)];
 }
 
+- (void)testPencilSupport API_AVAILABLE(ios(13.4)) {
+  FlutterViewController* vc = [[FlutterViewController alloc] initWithEngine:self.mockEngine
+                                                                    nibName:nil
+                                                                     bundle:nil];
+  XCTAssertNotNil(vc);
+
+  id mockPencilInteraction = OCMClassMock([UIPencilInteraction class]);
+
+  OCMStub([mockPencilInteraction preferredTapAction])
+      .andReturn(UIPencilPreferredActionShowColorPalette);
+
+  [vc pencilInteractionDidTap:mockPencilInteraction];
+
+  flutter::PointerData pointer_data;
+  pointer_data.Clear();
+  pointer_data.kind = flutter::PointerData::DeviceKind::kStylus;
+  pointer_data.signal_kind = flutter::PointerData::SignalKind::kStylusAuxiliaryAction;
+  pointer_data.preferred_auxiliary_stylus_action =
+      flutter::PointerData::PreferredStylusAuxiliaryAction::kShowColorPalette;
+
+  auto packet = std::make_unique<flutter::PointerDataPacket>(1);
+  packet->SetPointerData(/*index=*/0, pointer_data);
+
+  [[[self.mockEngine verify] ignoringNonObjectArgs] dispatchPointerDataPacket:std::move(packet)];
```

Comment:
`ignoringNonObjectArgs` is ignoring the value of the `packet` and accepting anything, so you aren't actually testing anything here except that `dispatchPointerDataPacket` was called at least once.  When I swap the action to `kSwitchEraser` it still passes.

This doesn't pass but something like:


```objc
  id mockPencilInteraction = OCMClassMock([UIPencilInteraction class]);

  OCMStub([mockPencilInteraction preferredTapAction])
      .andReturn(UIPencilPreferredActionShowColorPalette);

  XCTestExpectation* pointerDataPacketExpectation =
      [self expectationWithDescription:@"pointerDataPacket"];
  OCMStub(
      [self.mockEngine dispatchPointerDataPacket:std::make_unique<flutter::PointerDataPacket>(0)])
      .ignoringNonObjectArgs()
      .andDo(^(NSInvocation* invocation) {
        flutter::PointerData packet;
        [invocation getArgument:&packet atIndex:2];
        XCTAssertEqual(packet.kind, flutter::PointerData::DeviceKind::kStylus);
        XCTAssertEqual(packet.signal_kind,
                       flutter::PointerData::SignalKind::kStylusAuxiliaryAction);
        XCTAssertEqual(packet.preferred_auxiliary_stylus_action,
                       flutter::PointerData::PreferredStylusAuxiliaryAction::kShowColorPalette);
        [pointerDataPacketExpectation fulfill];
      });

  [vc pencilInteractionDidTap:mockPencilInteraction];

  [self waitForExpectationsWithTimeout:1 handler:nil];

  [mockPencilInteraction stopMocking];
```

Reply:
Hm, this still doesn't pass but it's closer, it needs to get the pointer data off the packet.
```objc
    std::unique_ptr<flutter::PointerDataPacket> packet;
    [invocation getArgument:&packet atIndex:2];
    flutter::PointerData pointer_data = packet->GetPointerData(0);

    XCTAssertEqual(pointer_data.kind, flutter::PointerData::DeviceKind::kStylus);
    XCTAssertEqual(pointer_data.signal_kind,
                   flutter::PointerData::SignalKind::kStylusAuxiliaryAction);
    XCTAssertEqual(pointer_data.preferred_auxiliary_stylus_action,
                   flutter::PointerData::PreferredStylusAuxiliaryAction::kShowColorPalette);
    [pointerDataPacketExpectation fulfill];
```

Reply:
I did some research just now - it looks like there's no such OCMock API for dereferencing smart pointers at all. I think OCMock only supports dereferencing raw pointers. If that's the case, `verify` maybe the only choice here for this test. 

Another idea is to break out the main logic into a helper function like `getAxiliaryActionPacket()` so it's easier to test.

Reply:
> I could be wrong, but I wasn't able to find any examples online. If that's the case, `verify` maybe the only choice here for this test.

How do you suggest getting `verify` working?  It would need to do some equality check.

Reply:
I was suggesting to just verify that it is called (like [this](https://github.com/flutter/engine/commit/a8a7534eabbad4eedbc3fa2963cdefd604541709#diff-9df3e6878a0f23383e1f2de991386935e435c5b77c863de75bd67a4ced38591dR697-R698)). 

For the equality check, i think we can factor out a helper function `getAxiliaryActionPacket()` and test that instead.